honeybee_vtk/data.py

```diff
@@ -0,0 +1,161 @@
+                    ' identifiers in HBJSON.'
+                )
+            return True
+
+        # if object_name is other than grid check that length of data matches the length
+        # of data in the model for that object.
+        elif self.object_name in AcceptedValues.names.value[:-1]:
+
+            if len(self.file_paths) == 0:
+                raise ValueError(
+                    'File path not found in the config file.'
+                )
+            elif len(self.file_paths) > 1:
+                raise ValueError(
+                    'Only one file path needs to be provided in order to load data on'
+                    f' {self.object_name}. Multiple files are provided in the config'
+                    ' file.'
+                )
+
+            file = open(self.file_paths[0], "r")
+            nonempty_line_count = len(
+                [line.strip("\n") for line in file if line != "\n"]
+            )
+            file.close()
+
+            if nonempty_line_count != len(
+                    model.get_modeldataset_from_string(self.object_name).data):
+                raise ValueError(
+                    'The length of data in the file does not match the number of'
+                    f' {self.object_name} objects in the model.'
+                )
+            return True
+
+        else:
+            raise ValueError(
+                ' object_name must be from one of these'
+                f' {AcceptedValues.names.value}. Instead got {self.object_name}.'
+            )
+
+
+class DataConfig(BaseModel):
+    """Config for the whole data config file."""
+
+    data: Dict[str, Data] = Field(
+        description='A dictionary to introduce data that you would like to mount.'
+        ' The key must be any text and the value must be DataConfig.'
+    )
+
+    def check_data(self, model: Model) -> bool:
+        return all([val.cross_check_data(model) for val in self.data.values()])
+
+
+def check_data_config(config_path: str, model: Model) -> Dict[str, DataConfig]:
+    """Validate the data config file.
+
+    Args:
+        config_path: File path to the data config file.
+        model: A honeybee-vtk model object.
+
+    Returns:
+        Data config as a dictionary.
+    """
+    # Check if path is valid
+    path = Path(config_path)
+    if not path.exists():
+        raise FileExistsError(
+            f'{config_path} is not a valid path.'
+        )
+
+    # Check if json is valid
+    try:
+        with open(config_path) as fh:
+            config = json.load(fh)
+    except json.decoder.JSONDecodeError:
+        raise TypeError(
+            'Not a valid json file.'
+        )
+
+    # Check if schema is valid
+    else:
+        json_obj = DataConfig.parse_file(path)
+        if json_obj.check_data(model):
+            return json_obj.dict(exclude_none=True)
+        else:
+            raise ValueError(
+                'Validation failed.'
+            )
```

Comment:
You only need the last check and it should raise an error if the file is not valid. the first check for loading them twice is redundant.

```suggestion
      config = DataConfig.parse_file(path)
      config .check_data(model)  # should raise an error if data is not valid
      return config.dict(exclude_none=True)
```